torch/_inductor/cudagraph_trees.py

```diff
@@ -1792,6 +1793,40 @@ def run(self, new_inputs: List[Tensor], function_id: FunctionID):
     def set_to_running_backward(self):
         self.running_forwards_with_pending_backwards = False
 
+    def _get_cuda_graph_recorded_tensor_checker(self) -> Callable[[Tensor], bool]:
+        return (
+            self.current_node._is_cuda_graph_recorded_tensor
+            if isinstance(self.current_node, CUDAGraphNode)
+            else lambda _: False
+        )
+
+    def new_warmup_node_id(self) -> int:
+        return next(self.warmup_node_counter)
+
+    def _update_non_cudagraph_managed_mutation(
+        self, function_id: FunctionID, inputs: List[Tensor]
+    ):
+        node_id = self._get_node_id()
+        if has_mutation_str := check_for_mutation(
+            self.ids_to_funcs[function_id],
+            inputs,
+            self._get_cuda_graph_recorded_tensor_checker(),
+        ):
+            perf_hint_log.warning(has_mutation_str)
+            self.non_cudagraph_managed_mutation_hint[node_id][function_id] = True
```

Comment:
So in the current state if you have :

```
torch.set_default_device('cuda')

@torch.compile(mode='reduce-overhead')
def foo(x):
    return x + 1

@torch.compile(mode='reduce-overhead')
def fee(y, z): 
    return z.add_(3)

for _ in range(3):
    torch.compiler.cudagraph_mark_step_begin()
    fee(torch.rand([4]), foo(torch.rand([4]))   
```

The first warmup node for `foo` and the subsequent CUDAGraphNode for `foo` will have different IDs, so we'll check twice. That's definitely fine, since the Warmup nodes are temporary I think it'd be more work than it's worth to avoid the subsequent second check. However, we should make sure that we are not emitting warning spam and warning twice which I think is what happens here. Maybe let's warn at most once per function id ? Can you add a test that checks this?

If i'm wrong here please ignore haha